scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java

```diff
@@ -389,6 +389,9 @@ public JobType getType() {
                 }
             }
 
+            //replace variables
+            commonPropertiesHolder.getVariables().putAll(replaceVariablesInJobVariablesMap(commonPropertiesHolder.getVariables(),replacementVariables));
```

Comment:
It is a bit hard to read. In the same line you 'putAll', and also replace  something.
I do not know what you can change.. but maybe you know.
Thanks